plugins/ua_config_file_based.c

```diff
@@ -0,0 +1,575 @@
+    if(fieldName != NULL)
+        printf("]\n");
+    return retval;
+}
+PARSE_JSON(DateTimeField) {
+    UA_Int64 dateTime_value;
+    Int64Field_parseJson(ctx, fieldName, &dateTime_value, NULL);
+    /* set server config field */
+    UA_DateTime *field = (UA_DateTime*)configField;
+    *field = (UA_DateTime)dateTime_value;
+    return UA_STATUSCODE_GOOD;
+}
+
+/*----------------------Enumerations------------------------*/
+PARSE_JSON(ApplicationTypeField) {
+    UA_UInt32 enum_value;
+    UInt32Field_parseJson(ctx, fieldName, &enum_value, NULL);
+    /* set server config field */
+    UA_ApplicationType *field = (UA_ApplicationType*)configField;
+    *field = (UA_ApplicationType)enum_value;
+    return UA_STATUSCODE_GOOD;
+}
+PARSE_JSON(RuleHandlingField) {
+    UA_UInt32 enum_value;
+    UInt32Field_parseJson(ctx, fieldName, &enum_value, NULL);
+    /* set server config field */
+    UA_RuleHandling *field = (UA_RuleHandling*)configField;
+    *field = (UA_RuleHandling)enum_value;
+    return UA_STATUSCODE_GOOD;
+}
+
+const parseJsonSignature parseJsonJumpTable[UA_SERVERCONFIGFIELDKINDS] = {
+    /* Basic Types */
+    (parseJsonSignature)Int64Field_parseJson,
+    (parseJsonSignature)UInt16Field_parseJson,
+    (parseJsonSignature)UInt32Field_parseJson,
+    (parseJsonSignature)StringField_parseJson,
+    (parseJsonSignature)LocalizedTextField_parseJson,
+    (parseJsonSignature)DoubleField_parseJson,
+    (parseJsonSignature)BooleanField_parseJson,
+    (parseJsonSignature)DurationField_parseJson,
+    (parseJsonSignature)DurationRangeField_parseJson,
+    (parseJsonSignature)UInt32RangeField_parseJson,
+
+    /* Advanced Types */
+    (parseJsonSignature)BuildInfo_parseJson,
+    (parseJsonSignature)ApplicationDescriptionField_parseJson,
+    (parseJsonSignature)StringArrayField_parseJson,
+    (parseJsonSignature)DateTimeField_parseJson,
+
+    /* Enumerations */
+    (parseJsonSignature)ApplicationTypeField_parseJson,
+    (parseJsonSignature)RuleHandlingField_parseJson,
+};
+
+static UA_StatusCode
+parseJSONConfig(UA_ServerConfig *config, UA_ByteString json_config) {
+    // Parsing json config
+    const char *json = (const char*)json_config.data;
+    cj5_token tokens[MAX_TOKENS];
+    cj5_result r = cj5_parse(json, (unsigned int)json_config.length, tokens, MAX_TOKENS, NULL);
+
+    ParseCtx ctx;
+    ctx.json = json;
+    ctx.result = r;
+    ctx.tokens = r.tokens;
+    ctx.tokensSize = r.num_tokens;
+    ctx.index = 3; // The first three tokens are ignored because they are known and not needed.
+
+    size_t serverConfigSize = (ctx.tokens[ctx.index-1].size/2);
+    for (size_t j = serverConfigSize; j > 0; j--) {
+        cj5_token tok = ctx.tokens[ctx.index];
+        switch (tok.type) {
+            case CJ5_TOKEN_STRING: {
+                char field[tok.size + 1];
+                unsigned int str_len = 0;
+                cj5_get_str(&ctx.result, (unsigned int)ctx.index, field, &str_len);
```

Comment:
The if-else horror can be replaced with a lookup table.
With the field name, the data type, an offset to the config member and an additional method pointer if we have a special way of handling the parameter outside of just decoding.